lib/request.js

```diff
@@ -8,19 +8,31 @@ exports.get = (options, config, cb) => {
     return makeRequest(options, config, cb);
 };
 
-exports.post = (options, config, cb) => {
+exports.post = (options, config, cb, ctx) => {
+    if (ctx && ctx.riskRttLogs) {
+        ctx.riskRttLogs += `request.post:${Date.now() - ctx.startRiskRtt};`;
+    }
     options.method = 'POST';
     if (!options.headers['content-type'] && !options.headers['Content-Type']) {
         options.headers['content-type'] = 'application/json';
     }
-    return makeRequest(options, config, cb);
+    return makeRequest(options, config, cb, ctx);
 };
 
-function makeRequest(options, config, cb) {
+function makeRequest(options, config, cb, ctx) {
+    if (ctx && ctx.riskRttLogs) {
+        ctx.riskRttLogs += `makeRequest:${Date.now() - ctx.startRiskRtt};`;
+    }
     if (options.url && options.url.startsWith('https://')) {
         options.agent = config.agent || httpsKeepAliveAgent;
     } else {
         options.agent = new http.Agent();
     }
+    if (ctx && ctx.riskRttLogs) {
+        ctx.riskRttLogs += `beforeP:${Date.now() - ctx.startRiskRtt};`;
+    }
     p(options, cb);
+    if (ctx && ctx.riskRttLogs) {
+        ctx.riskRttLogs += `afterP:${Date.now() - ctx.startRiskRtt};`;
```

Comment:
Nope!
This is async!
You need to measure it from the CB of p

Reply:
I've added measurments to this cb too (in the call to request.post(...) in the `httpc` file). This `afterP ` is just an extra log
![image](https://user-images.githubusercontent.com/54812214/145444271-2fbd71e3-ee77-44a5-a555-55cd411c469e.png)